### DISCUSSION TITLE
SetNewWindowDefaultPos and some others (probably all deprecated)

### DISCUSSION THREAD

**Discussion by heysokam** [ocornut/imgui#5560] [New Users Build/Link/Run issues ONLY!]
Hello. I'm trying to compile the demo at:   
https://github.com/RandyGaul/qu3e

I haven't tested their compiling, because I never used CMake, and don't know how to use it. So its easier for me to just compile with the latest libraries and my usual toolchain instead, (SCons and gcc on linux, if that's important)

But the demo is using some really old implementation of imgui, and I don't know how to update it so that It can be built with the latest versions _(first time user of imgui)_

These are the problematic lines that I can see at the moment _(there might be more, but the LSP shows these)_:
```c
// No member named SetNewWindowDefaultPos in namespace ImGui
ImGui::SetNewWindowDefaultPos(ImVec2(float(w - 300 - 30), 30));

// No viable conversion from ImVec2 to ImGuiWindowFlags
ImGui::Begin("q3Scene Settings", NULL, ImVec2(300, 225));

ImGuiIO &io = ImGui::GetIO();
// No member named PixelCenterOffset in ImGuiIO
io.PixelCenterOffset = 0.0f;  // Align OpenGL texels
// No member named RenderDrawListsFn in ImGuiIO
io.RenderDrawListsFn = ImImpl_RenderDrawLists;

// No member named GetDefaultFontData in namespace ImGui
ImGui::GetDefaultFontData(NULL, NULL, &png_data, &png_size);
```

I cannot find them in <imgui.h>, or none of the other included files from that one. Not even commented out.
How can I figure out if they were deprecated, or if I'm just missing some extra header?

**Comment by ocornut** [comment]
Search for those names in the codebase you’ll find mentions in the API BREAKING CHANGES section in imgui.cpp
